lib/std/os.zig

```diff
@@ -3111,6 +3111,67 @@ pub fn send(
     return sendto(sockfd, buf, flags, null, 0);
 }
 
+pub const SendFileError = error{
+    /// There was an unspecified error while reading from infd.
+    InputOutput,
+
+    /// There was insufficient memory for reading from infd.
+    NoMem,
```

Comment:
I'm surprised (and disappointed) this error is even possible; the kernel should be able to make this "just work" even if it is forced to use a small stack-based buffer due to heap memory pressure.

By convention, ENOMEM in zig turns into one of these two things:
 * `error.OutOfMemory` if it is userspace heap memory that is exhausted
 * `error.SystemResources` if it is kernel space memory or other resources that are unavailable.

In this case I believe it would be `error.SystemResources`.

Reply:
Reading through the kernel sources, this actually comes from files that have a mandatory lock, and the underlying splice() cannot return ENOMEM.

Reply:
@shawnl Thanks, I can update the comment. I'll likely take @andrewrk's recommendation of using `error.SystemResources` in this case.